cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -183,13 +186,28 @@ public final ProgramController run(final Program program, final ProgramOptions o
       final ProgramOptions options = addArtifactPluginFiles(oldOptions, localizeResources,
                                                             DirUtils.createTempDir(tempDir));
 
+      final List<String> additionalClassPaths = new ArrayList<>();
+      List<String> newCConfExtraJars = new ArrayList<>();
+      // Add extra jars set in cConf to additionalClassPaths and localizeResources
+      for (URI jarURI : CConfigurationUtil.getExtraJars(cConf)) {
+        String protocol = jarURI.getScheme();
```

Comment:
should be called `scheme`, not `protocol`. Also, you can move this as an inline call in line 199